odbc_fdw.c

```diff
@@ -1403,11 +1383,11 @@ odbcIterateForeignScan(ForeignScanState *node)
         int k;
         bool found;
 
-        #if !defined (TMPHACK)
-        /* Allocate memory for the masks */
+        /* Allocate memory for the masks in a memory context that
+           persists between IterateForeignScan calls */
+        prev_context = MemoryContextSwitchTo(estate->es_query_cxt);
```

Comment:
to be honest, I need to review the documentation to know more about memory context, exectuor states, scope and lifecycle of those memory regions.